src/input/source/led/multicolor_chassis.rs

```diff
@@ -0,0 +1,167 @@
+pub struct MultiColorChassis {
+    multi_intensity_path: PathBuf,
+    multi_index_map: Vec<IndexType>,
+    fixed_color: Option<LedFixedColor>,
+}
+
+impl MultiColorChassis {
+    /// Create a new MultiColorChassis source device with the given udev
+    /// device information
+    pub fn new(
+        device_info: UdevDevice,
+        fixed_color: Option<LedFixedColor>,
+    ) -> Result<Self, Box<dyn Error + Send + Sync>> {
+        let multi_intensity_path =
+            PathBuf::from(device_info.syspath().as_str()).join("multi_intensity");
+        if !multi_intensity_path.exists() {
+            return Err(Box::new(MultiColorChassisError::MultiIntensityNotFound(
+                multi_intensity_path,
+            )));
+        }
+
+        let multi_index = PathBuf::from(device_info.syspath().as_str()).join("multi_index");
+        let contents = std::fs::read_to_string(multi_index)
+            .map_err(|err| Box::new(MultiColorChassisError::MultiIndexError(err)))?;
+        let multi_index_strings = contents
+            .split_whitespace()
+            .into_iter()
+            .map(|str| String::from(str))
+            .collect::<Vec<String>>();
+        let mut multi_index_map = Vec::<IndexType>::with_capacity(multi_index_strings.len());
+        for idx in multi_index_strings.iter() {
+            match idx.as_str() {
+                "rgb" => multi_index_map.push(IndexType::RGB),
+                _ => {
+                    return Err(Box::new(MultiColorChassisError::UnsupportedIndexType(
+                        idx.clone(),
+                    )))
+                }
+            }
+        }
+
+        let result = Self {
+            fixed_color,
+            multi_intensity_path,
+            multi_index_map,
+        };
+
+        // Immediatly set the user-defined color if one is provided
+        if let Some(color) = result.fixed_color.as_ref() {
+            result.write_color(color.r, color.g, color.b)?
+        };
+
+        Ok(result)
+    }
+
+    fn write_color(&self, r: u8, g: u8, b: u8) -> Result<(), Box<dyn Error + Send + Sync>> {
+        let contents = std::fs::read_to_string(self.multi_intensity_path.as_path())
+            .map_err(|err| Box::new(MultiColorChassisError::MultiIndexError(err)))?;
+        let contents = (self
+            .multi_index_map
+            .iter()
+            .zip(contents.split_whitespace().into_iter())
+            .map(|(index_type, index_value)| match index_type {
+                IndexType::RGB => {
+                    (((r as u32) << 16u32) | ((g as u32) << 8u32) | ((b as u32) << 0u32))
+                        .to_string()
+                }
+                _ => String::from(index_value),
+            })
+            .collect::<Vec<String>>())
+        .join(" ");
+        Ok(
+            std::fs::write(self.multi_intensity_path.as_path(), contents)
+                .map_err(|err| Box::new(MultiColorChassisError::MultiIntensityUpdateError(err)))?,
+        )
```

Comment:
To fix clippy warnings:

```suggestion
        let values = self
            .multi_index_map
            .iter()
            .zip(contents.split_whitespace())
            .map(|(index_type, _index_value)| match index_type {
                IndexType::Rgb => {
                    (((r as u32) << 16u32) | ((g as u32) << 8u32) | (b as u32)).to_string()
                }
            })
            .collect::<Vec<String>>();
        let contents = values.join(" ");

        std::fs::write(self.multi_intensity_path.as_path(), contents)
            .map_err(|err| Box::new(MultiColorChassisError::MultiIntensityUpdateError(err)))?;

        Ok(())
```